src/utils/format-string.ts

```diff
@@ -1,18 +1,21 @@
 import capitalizeFirstLetter from "../utils/capitalize-first-letter"
 
-const formatString = (str: string): string => {
-  str = str.trim()
-  str = str
-    .replace(/,\s*/g, ", ")
-    .replace(/\.(\s*)/g, ". ")
-    .replace(/\?(\s*)/g, "? ")
-    .replace(/!(\s*)/g, "! ")
+/** Checks if punctuation mark ends without a whitespace at the end.
+ * This represents a paragraph
+ */
+const checkParagraph: RegExp = /[.?!](?!\s)/g
```

Comment:
wouldn't this example fail?
```
this is a paragraph.

this should be another paragraph, but because it matches on all whitespaces it is not one.
```

Reply:
The regex is actually to check if there's a missing paragraph, it was written mainly for the /horoscope command since the API throws us the text with all the paragraphs missing. Think my comment was not very clear, oooops.

e.g.

Use the start of the week as an excuse to get your life together when the sun in Taurus sextiles Saturn in Pisces on Monday. You’re in the process of changing your life, starting with developing better habits. Create some grounding routines to help boost your empowerment, **Libra.Take** relationships of all kinds to the next level for stability and structure when the new moon is in Taurus and your intimacy zone on Tuesday. Let go of your hang-ups and embrace being known. It’s a great time to share financial burdens with a partner as you truly become a team. Good **luck!Friday** is ideal for heart-to-heart conversations when the moon in Gemini sextiles Mercury in Aries. You’re ready to open your heart to someone special and level up your relationships. This is a great aspect for dreaming big and letting your feelings be known.

(bolded missing paragraphs)

It's made into a util because in future if other APIs throw missing paragraphs at us we can just reuse this and add the paragraphs back

Reply:
> ```
> this is a paragraph.
> 
> this should be another paragraph, but because the regex matches on all whitespaces it is not one.
> ```

So since this is already split properly into paragraphs we can just ignore it

Reply:
ah ha, i see. LGTM!